cmake/CMakeLists.txt

```diff
@@ -123,6 +123,10 @@ else()
   endif()
   add_executable(lammps ${ALL_SOURCES})
   set_target_properties(lammps PROPERTIES OUTPUT_NAME ${LAMMPS_BINARY})
+  if(Kokkos_ENABLE_CUDA)
+    # The Kokkos nvcc wrapper does not like -std=gnu++11 and wants -std=c++11
+    set_target_properties(lammps PROPERTIES CXX_EXTENSIONS OFF)
```

Comment:
Are we using gnu extension anywhere? Otherwise could just make that the default.

Reply:
not that i know of. we used to recommend `-std=c++11` previously anyway. and using GNU extensions would be non-portable. so i'll take out the condition.

Reply:
Ok, I would use `set(CMAKE_CXX_EXTENSIONS OFF)` where we set `CMAKE_CXX_STANDARD` above, so that it applies for all target in this project.